### DISCUSSION TITLE
devcontainer support with linting?

### DISCUSSION THREAD

**Discussion by smos08** [tomaciazek/vscode-ansible#39] [Q&A] [answered]
does this work with vscode devcontainers?  i am able to get everything working except linting does not seem to show in the problems area or underline in the editor.  does anyone have this extension working with linting inside a devcontainer?

**Comment by tomaciazek** [comment, marked as answer]
Hi @smos08, I'm actually running it in a devcontainer on a daily basis, so it's well tested 😉 Make sure that `ansible-lint` is installed in your container and that VS Code knows how to find it (there are configuration options for that, if necessary).

There could also be an issue in your playbooks that `ansible-lint` doesn't handle gracefully and crashes without output. You should check the output of Ansible Server. Also, try running the linter from the command-line and see what it returns.

**Reply by smos08** [reply]
Hey @tomaciazek, first off thank you so much for this great add on.  i assumed it should be working fine with devcontainers and that the issue is on my end with setup.  i am able to attach to the dev container and manually run ansible-lint with success. i have also tried to set the ansible-lint path explicitly in your plugin which also does not change the behavior.  i also confirmed the playbook i am checking does in fact register issues since i am able to see the linting when not using a devcontainer.  i have attached my devcontainer and dockerfile here, maybe something stands out?  or can you post the config you are using?

.devcontainer
```
// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
// https://github.com/microsoft/vscode-dev-containers/tree/v0.191.1/containers/debian
{
	"name": "Debian",
	"build": {
		"dockerfile": "Dockerfile",
		// Update 'VARIANT' to pick an Debian version: bullseye, buster, stretch
		"args": { "VARIANT": "buster" }
	},

	// Set *default* container specific settings.json values on container create.
	"settings": {},

	// Add the IDs of extensions you want installed when the container is created.
	"extensions": [
		"tomaciazek.ansible",
		"redhat.vscode-yaml",
		"ms-azuretools.vscode-docker",
		"ms-python.python"
	],

	"remoteUser": "root"
}
```

dockerfile 

```
# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.191.1/containers/debian/.devcontainer/base.Dockerfile

# [Choice] Debian version: bullseye, buster, stretch
ARG VARIANT="buster"
FROM mcr.microsoft.com/vscode/devcontainers/base:0-${VARIANT}

ENV ANSIBLE_HOST_KEY_CHECKING false


RUN echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu trusty main" >> /etc/apt/sources.list && \
   sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 93C4A3FD7BB9C367


# ** [Optional] Uncomment this section to install additional packages. **
RUN sudo apt-get update && export DEBIAN_FRONTEND=noninteractive && \
   sudo apt-get -y install gcc && \
   sudo apt-get -y install python3-pip && \
   pip3 install --upgrade pip

RUN pip3 install --upgrade pip
RUN pip3 install setuptools wheel
RUN pip3 install ansible-lint[community,yamllint] "ansible>=2.9,<2.10"
```

**Reply by tomaciazek** [reply]
I've tried your setup. Indeed the linting doesn't work, and here's the output of Ansible Server (you can find it under `Output > Ansible Server`):
```
ansible-lint: error: argument -f: invalid choice: 'codeclimate' (choose from 'rich', 'plain', 'rst')
```

Looks like that line installs an old version of `ansible-lint`, which doesn't support the `codeclimate` output format:
```
RUN pip3 install ansible-lint[community,yamllint] "ansible>=2.9,<2.10"
```
Try something like:
```
RUN pip3 install ansible-lint[yamllint] "ansible>=2.9,<2.10"
```

**Reply by smos08** [reply]
that worked like a charm! thank you for finding it.  

i also didn't know about the "output" area for plugin info so i have also learned something additional.  

thank you again for the plugin and digging into this with me, truly grateful
